src/pow.cpp

```diff
@@ -71,6 +71,54 @@ unsigned int CalculateNextWorkRequired(const CBlockIndex* pindexLast, int64_t nF
     return bnNew.GetCompact();
 }
 
+// Check that on difficulty adjustments, the new difficulty does not increase
+// or decrease beyond the permitted limits.
+bool PermittedDifficultyTransition(const Consensus::Params& params, int64_t height, uint32_t old_nbits, uint32_t new_nbits)
+{
+    if (params.fPowAllowMinDifficultyBlocks) return true;
+
+    if (height % params.DifficultyAdjustmentInterval() == 0) {
+        int64_t smallest_timespan = params.nPowTargetTimespan/4;
+        int64_t largest_timespan = params.nPowTargetTimespan*4;
+
+        const arith_uint256 bnPowLimit = UintToArith256(params.powLimit);
```

Comment:
while we're here, what does `bn` actually refer to? :grimacing: 

wouldn't mind tossing the old naming schemes for this

Reply:
@instagibbs In the old Satoshi-era naming convention, it referred to "bignum" (there was a `CBigNum` wrapper around the OpenSSL `BIGNUM` type, which was used for various big integer operations, both inside script and for PoW purposes).

Reply:
Done.